### PR TITLE
Adding component tests for Multi Stacked Bar Chart

### PR DESCRIPTION
**Pull request by srmukher**
Adding component tests for Multi Stacked Bar Chart

**Improved code coverage from 60.46% to 81.39%:**

Master (code coverage of  60.46%):
image

Improved (code coverage of 81.39%):
image

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/c0214bcb7d9369db75eb842adf4253893bc0e160'>c0214bcb7d9369db75eb842adf4253893bc0e160</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/docs/TestPlans/MultiStackedBarChart/componentTests.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+**Multi stacked bar Chart – Component test plan**
```
Comment (AtishayMsft) [inline packages/react-charting/docs/TestPlans/MultiStackedBarChart/componentTests.md:1]: Add a test case to show placeholder bar.

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 2]
```diff
@@ -253,6 +256,7 @@ export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarCh
         >
           <rect
             key={index}
+            id={getId('_MSBC_rect_')}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:259]: You can call getId only once and append the key to it to make it unique.
Similar approach is followed in other charts

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+      expect(getById(container, /_MSBC_bar/i)).toHaveLength(data.length);
+      expect(getById(container, /_MSBC_rect/i)).toHaveLength(12);
+      const rect1: number = +rect[10].getAttribute('width')!.split('%')[0];
+      const rect2: number = +rect[11].getAttribute('width')!.split('%')[0];
+      expect(rect1 + rect2).toEqual(100);
+    },
+  );
+
+  testWithWait(
+    'Should render the multi stacked bar with the given variant part-to-whole',
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx:164]: absolute-scale

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    MultiStackedBarChart,
+    { data, variant: 'absolute-scale' },
+    container => {
+      const rect = getById(container, /_MSBC_rect/i);
+      // Assert
+      expect(getById(container, /_MSBC_bar/i)).toHaveLength(data.length);
+      expect(getById(container, /_MSBC_rect/i)).toHaveLength(12);
+      const rect1: number = +rect[10].getAttribute('width')!.split('%')[0];
+      const rect2: number = +rect[11].getAttribute('width')!.split('%')[0];
+      expect(rect1 + rect2).not.toEqual(100);
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx:174]: We can make the test more robust. Here the biggest data point group has 100% width and other groups have width as per the values.

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx (hunk 3 above)
Comment (srmukher) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx:164]: done

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx (hunk 4 above)
Comment (srmukher) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChartRTL.test.tsx:174]: done

File: packages/react-charting/docs/TestPlans/MultiStackedBarChart/componentTests.md (hunk 1 above)
Comment (srmukher) [reply packages/react-charting/docs/TestPlans/MultiStackedBarChart/componentTests.md:1]: done

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 5]
```diff
@@ -247,6 +256,7 @@ export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarCh
         >
           <rect
             key={index}
+            id={this._barId + index}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:259]: Use prefix-{index} format

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 6]
```diff
@@ -282,14 +292,30 @@ export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarCh
       if (data.chartData!.length === 0) {
         bars.push(
           <g key={0} className={this._classNames.noData} onClick={this._redirectToUrl.bind(this, href)}>
-            <rect key={0} x={'0%'} y={0} width={'100%'} height={barHeight} fill={palette.neutralLight} />
+            <rect
+              key={0}
+              id={this._barIdEmptyPartToWhole}
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:297]: These are placeholder bars, not empty bars.

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 7]
```diff
@@ -72,6 +75,9 @@ export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarCh
     this._onBarLeave = this._onBarLeave.bind(this);
     this._calloutId = getId('callout');
     this._emptyChartId = getId('_MSBC_empty');
+    this._barId = getId('_MSBC_rect_');
+    this._barIdEmptyPartToWhole = getId('_MSBC_rect_partToWhole_');
+    this._barIdEmpty = getId('_MSBC_rect_empty');
   }
 
   public render(): JSX.Element {
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:83]: With _isChartEmpty added, are we ever showing the placeholder bars now?

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx (hunk 6 above)
Comment (srmukher) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:297]: done

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx (hunk 5 above)
Comment (srmukher) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:259]: done

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx [hunk 8]
```diff
@@ -102,7 +108,11 @@ export class MultiStackedBarChartBase extends React.Component<IMultiStackedBarCh
           this.props.hideDenominator![index],
           this.props.href,
         );
-        return <div key={index}>{singleChartBars}</div>;
+        return (
+          <div key={index} id={getId('_MSBC_bar_')}>
```
[the remark this answers is not in the document]
Comment (srmukher) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:112]: done

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: change/@fluentui-react-charting-b86a16d6-a1aa-4d4f-bb2c-b1632ca07a8d.json [hunk 9]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-charting-b86a16d6-a1aa-4d4f-bb2c-b1632ca07a8d.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=28165&commitId=ee77ff8412ec514055e4d63644b0d73dea08a4ac&env=prod&clientType=FLUENTUI)


  <details><summary><b>Panel</b> 8 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Panel.Custom anchored left.default.chromium.png|24| Changed|
|Panel.Large fixed right.default.chromium.png|24| Changed|
|Panel.Extra large right.default.chromium.png|24| Changed|
|Panel.SearchBox and Right Panel - RTL.default.chromium.png|28| Changed|
|Panel.With no header, close button.default.chromium.png|24| Changed|
|Panel.Custom.default.chromium.png|24| Changed|
|Panel.SearchBox and Right Panel.default.chromium.png|28| Changed|
|Panel.With footer at the bottom.default.chromium.png|24| Changed|

</details>


  <details><summary><b>react-charting-AreaChart</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|react-charting-AreaChart.Custom Accessibility.default.chromium.png|11| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

File: packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx (hunk 7 above)
Comment (srmukher) [reply packages/react-charting/src/components/StackedBarChart/MultiStackedBarChart.base.tsx:83]: yes, we are now showing as discussed offline

**Review by AtishayMsft — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Pull request merged**